classes/Employee.php

```diff
@@ -316,9 +316,6 @@ public function getByEmail($email, $plaintextPassword = null, $activeOnly = true
         }
 
         $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow($sql);
-        if (!$result) {
```

Comment:
This line being removed maybe it then raises a notice on line 326 below? Because line 326 does this:
```
$passwordHash = $result['passwd'];
```

So we access an unset key.

Maybe instead we should create an invalid array?
```
if (!$result) {
    // create fake result to make sure computing time does not allow password enumeration, see https://github.com/PrestaShop/PrestaShop/pull/31497
    $result = ['passwd' => '123456'];
}

Reply:
This is not needed as `$result['passwd']` is equal `null`. `null` is as fine in this case. From my tests the behavior is correct and it does not raise any exception. :-)

Reply:
You are right. :-)